src/board.py

```diff
@@ -48,24 +48,33 @@ def __init__(self):
     def board_coordinates(cls, file_rank: str) -> tuple[int, int]:
         return cls.rank_to_index[file_rank[1]], cls.file_to_index[file_rank[0]]
 
-    def __setitem__(self, key, value):
+    def __setitem__(self, key: str, value: Piece):
         i, j = self.board_coordinates(key)
-        self.matrix[i][j] = value
+        self.board[i][j] = value
 
-    def __getitem__(self, key):
+    def __getitem__(self, key: str):
         i, j = self.board_coordinates(key)
-        return self.matrix[i][j]
+        return self.board[i][j]
+
+    def __delitem__(self, key: str):
+        i, j = self.board_coordinates(key)
+        del self.board[i][j]
+
+    def __iter__(self):
+        return iter(self.board)
+
+    def __contains__(self, value: Piece):
+        return any(value in rank for rank in self.board)
```

Comment:
Please add docstrings and rename `value` with `piece` for better clarity on the context of the methods.